src/modules/cpu/kernel/tensor_sum.hpp

```diff
@@ -0,0 +1,842 @@
+                {
+                    sum += (Rpp32u)(*srcPtrTemp);
+                    srcPtrTemp++;
+                }
+                srcPtrRow += srcDescPtr->strides.hStride;
+            }
+#if __AVX2__
+            rpp_simd_store(rpp_store8_u32_to_u32_avx, sumAvx, &psum);
+#endif
+            for(int i = 0; i < 4; i++)
+                sum += (sumAvx[i] + sumAvx[i + 4]);
+
+            tensorSumArr[batchCount] = (Rpp64u)sum;
+        }
+
+        // Tensor Sum without fused output-layout toggle 3 channel (NCHW)
+        else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp64u sum;
+            Rpp32u sumR = 0, sumG = 0, sumB = 0;
+            Rpp32u sumAvxR[8] = {0};
+            Rpp32u sumAvxG[8] = {0};
+            Rpp32u sumAvxB[8] = {0};
+
+            Rpp8u *srcPtrRowR, *srcPtrRowG, *srcPtrRowB;
+            srcPtrRowR = srcPtrChannel;
+            srcPtrRowG = srcPtrRowR + srcDescPtr->strides.cStride;
+            srcPtrRowB = srcPtrRowG + srcDescPtr->strides.cStride;
+#if __AVX2__
+            __m256i psumR = _mm256_setzero_si256();
+            __m256i psumG = _mm256_setzero_si256();
+            __m256i psumB = _mm256_setzero_si256();
+#endif
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtrTempR, *srcPtrTempG, *srcPtrTempB;
+                srcPtrTempR = srcPtrRowR;
+                srcPtrTempG = srcPtrRowG;
+                srcPtrTempB = srcPtrRowB;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+#if __AVX2__
+                    __m256i p[6];
+                    rpp_simd_load(rpp_load48_u8pln3_to_u32pln3_avx, srcPtrTempR, srcPtrTempG, srcPtrTempB, p);
+                    compute_sum_48_host(p, &psumR, &psumG, &psumB);
+#endif
+                    srcPtrTempR += vectorIncrementPerChannel;
+                    srcPtrTempG += vectorIncrementPerChannel;
+                    srcPtrTempB += vectorIncrementPerChannel;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    sumR += (Rpp32u)(*srcPtrTempR);
+                    sumG += (Rpp32u)(*srcPtrTempG);
+                    sumB += (Rpp32u)(*srcPtrTempB);
+                    srcPtrTempR++;
+                    srcPtrTempG++;
+                    srcPtrTempB++;
+                }
+                srcPtrRowR += srcDescPtr->strides.hStride;
+                srcPtrRowG += srcDescPtr->strides.hStride;
+                srcPtrRowB += srcDescPtr->strides.hStride;
+            }
+#if __AVX2__
+            rpp_simd_store(rpp_store8_u32_to_u32_avx, sumAvxR, &psumR);
+            rpp_simd_store(rpp_store8_u32_to_u32_avx, sumAvxG, &psumG);
+            rpp_simd_store(rpp_store8_u32_to_u32_avx, sumAvxB, &psumB);
+#endif
+            for(int i = 0; i < 4; i++)
```

Comment:
unrolling this loop might give better performance

Reply:
Incorporated this change